src/editor/utils/index.js

```diff
@@ -129,9 +130,13 @@ export const deepCopy = object => {
   return obj
 }
 
-export const loadImage = url => {
-  const image = new Image()
+export const loadImage = async (url, detectContentType) => {
+  if (detectContentType) {
+    const isImage = await checkImageContentType(url)
+    if (!isImage) throw new Error('not an image')
```

Comment:
I think you use `throw new Error()` just want to make the promise(loadImage returned) rejected so that we can catch it in `.catch()` block. but why don't you use `Promise.reject()` directly?

```javascript
if (!isImage) return Promise.reject(new Error('not an image'))
```

Reply:
Because it's redundant but I'll change it if you want.

Reply:
Another problem: `![](https://google.com/404)`

`return Promise.reject(new Error('not an image'))` crahses the renderer process while `throw new Error('not an image')` does not...

Reply:
>return Promise.reject(new Error('not an image')) crahses the renderer process while throw new Error('not an image') does not...

I run the codes locally, but it does not crash, can you show me the error message when it crashed?

Reply:
vscode debugger step into it: `not an image`.  Maybe `Promise.reject(new Error('not an image'))` needs different error handling?

Reply:
I think `return Promise.reject(new Error('not an image'))` and `throw new Error('not an image')` both will make the promise(return by async function) rejected.

They are all will be caught in `.catch` block.